controllers/space/space_controller_test.go

```diff
@@ -1029,6 +1031,165 @@ func TestUpdateSpaceTier(t *testing.T) {
+		s := spacetest.NewSpace("oddity",
+			spacetest.WithTierName(basicTier.Name),
+			spacetest.WithSpecTargetCluster("member-1"),
+			spacetest.WithStatusTargetCluster("member-1"), // already provisioned on a target cluster
+			spacetest.WithFinalizer())
+		nstmplSet := nstemplatetsettest.NewNSTemplateSet("oddity",
+			nstemplatetsettest.WithReferencesFor(basicTier,
+				// include pre-existing users with role...
+				nstemplatetsettest.WithSpaceRole("admin", adminMUR.Name),
+				nstemplatetsettest.WithSpaceRole("viewer", viewerMUR.Name),
+			),
+			nstemplatetsettest.WithReadyCondition(),
+		)
+		// ...and their corresponding space bindings
+		sb1 := spacebindingtest.NewSpaceBinding(adminMUR.Name, s.Name, "admin", "signupAdmin")
+		sb2 := spacebindingtest.NewSpaceBinding(viewerMUR.Name, s.Name, "viewer", "signupViewer")
+
+		// and a SpaceBinding for John as an Admin on the Space
+		johnMUR := murtest.NewMasterUserRecord(t, "john")
+		sb3 := spacebindingtest.NewSpaceBinding(johnMUR.Name, s.Name, "admin", "signupJohn")
+
+		hostClient := test.NewFakeClient(t, s, johnMUR, adminMUR, sb1, viewerMUR, sb2, sb3, basicTier)
+		member1Client := test.NewFakeClient(t, nstmplSet)
+		member1 := NewMemberClusterWithClient(member1Client, "member-1", corev1.ConditionTrue)
+		member2 := NewMemberCluster(t, "member-2", corev1.ConditionTrue)
+
+		ctrl := newReconciler(hostClient, member1, member2)
+
+		// when
+		res, err := ctrl.Reconcile(context.TODO(), requestFor(s))
+
+		// then
+		require.NoError(t, err)
+		assert.True(t, res.Requeue) // expect a requeue since the NSTemplateSet was updated
+		// Space should be in "updating" state while the
+		spacetest.AssertThatSpace(t, test.HostOperatorNs, s.Name, hostClient).
+			HasConditions(spacetest.Updating())
+		// NSTemplateSet should have an spaceRoles entry for the `mur`
+		nstemplatetsettest.AssertThatNSTemplateSet(t, test.MemberOperatorNs, nstmplSet.Name, member1Client).
+			HasSpaceRoleForUser("basic-admin-123456new", johnMUR.Name).    // entry added for user as admin
+			HasNoSpaceRoleForUser("basic-viewer-123456new", johnMUR.Name). // but not added as viewer
+			HasSpaceRoleForUser("basic-admin-123456new", adminMUR.Name).   // unchanged
+			HasSpaceRoleForUser("basic-viewer-123456new", viewerMUR.Name). // unchanged
+			HasConditions(nstemplatetsettest.Provisioned())                // not changed by the SpaceController, but will be by the NSTemplateSetController
+	})
+
+	t.Run("remove user with admin role", func(t *testing.T) {
+		// given a MUR, a Space and its NSTemplateSet resource...
+		s := spacetest.NewSpace("oddity",
+			spacetest.WithTierName(basicTier.Name),
+			spacetest.WithSpecTargetCluster("member-1"),
+			spacetest.WithStatusTargetCluster("member-1"), // already provisioned on a target cluster
+			spacetest.WithFinalizer())
+		adminMUR := murtest.NewMasterUserRecord(t, "jack")
+		viewerMUR := murtest.NewMasterUserRecord(t, "jeff")
+		johnMUR := murtest.NewMasterUserRecord(t, "john")
```

Comment:
it looks like that this set of MURs (plus maybe some additional resources like the `Space` as well as the `SpaceBinding`) are used in all newly added tests - you could instantiate them only once in the parent test and just reuse them. This would make the test much shorter and cleaner.

Reply:
ok, let me check

Reply:
done in 9d4b599